src/table.js

```diff
@@ -542,15 +550,81 @@ export function getTypeValidator(colType) {
   }
 }
 
+export function coerceToType(value, type) {
+  switch (type) {
+    case "string":
+      return typeof value === "string" || value == null
+        ? value
+        : String(value);
+    case "boolean":
+      if (typeof value === "string") {
+        const trimValue = value.trim();
+        return trimValue === "true" ? true : trimValue === "false" ? false : null;
+      }
+      return typeof value === "boolean" || value == null
+        ? value
+        : Boolean(value);
+    case "bigint":
+      return typeof value === "bigint"
+        ? value
+        : value === 0 || value === true || value === false
+        ? // eslint-disable-next-line no-undef
+          BigInt(value)
+        : !value || isNaN(value) || !Number.isInteger(+value)
+        ? NaN
+        : // eslint-disable-next-line no-undef
+          BigInt(value);
+    case "integer": // not a target type for coercion, but can be inferred
+    case "number": {
+      return typeof value === "number"
+        ? value
+        : value == null || value === ""
+        ? NaN
+        : Number(value);
+    }
+    case "date": {
+      if (value instanceof Date) return value;
+      const trimValue = typeof value === "string" ? value.trim() : value;
+      return new Date(trimValue);
```

Comment:
The behavior of the Date constructor is not consistent across browsers; some browsers (Chrome) are much more lenient than others (Safari). Maybe it’s okay to be as lenient as possible here, since we do a regex test when computing the inferred types for strings… but potentially we could repeat that regex test here, or perhaps use [isoformat’s parse](https://github.com/mbostock/isoformat#parsedate-fallback) (which we already use elsewhere) to only allow ISO 8601 formatted dates to be coerced.

Reply:
I like the idea of being as lenient as we can, so went with repeating the regex test here instead of using isoformat's parse (and I also revised the regex slightly after testing in Safari). But then, thinking that we want to return `Invalid Date` objects for anything that fails coercion, I added the fallback of returning new Date(""), which seems a little funky. (The `value &&` was needed because otherwise null gets coerced to Wed Dec 31 1969 16:00:00
GMT-0800 (Pacific Standard Time))